synthetics-stats/src/handlers/orders.ts

```diff
@@ -0,0 +1,154 @@
+import { Order, Transaction } from "../../generated/schema";
+import { BigInt } from "@graphprotocol/graph-ts";
+import { EventData } from "../utils/eventData";
+
+export function handleOrderCreated(
+  eventData: EventData,
+  transaction: Transaction
+): Order {
+  let key = eventData.getBytes32Item("key")!.toHexString();
+
+  let order = new Order(key);
+
+  order.account = eventData.getAddressItemString("account")!;
+  order.receiver = eventData.getAddressItemString("receiver")!;
+  order.callbackContract = eventData.getAddressItemString("callbackContract")!;
+  order.marketAddress = eventData.getAddressItemString("market");
+  order.swapPath = eventData.getAddressArrayItemString("swapPath")! || [];
+  order.initialCollateralTokenAddress = eventData.getAddressItemString(
+    "initialCollateralToken"
+  )!;
+  order.sizeDeltaUsd = eventData.getUintItem("sizeDeltaUsd")!;
+  order.initialCollateralDeltaAmount = eventData.getUintItem(
+    "initialCollateralDeltaAmount"
+  )!;
+  order.triggerPrice = eventData.getUintItem("triggerPrice");
+  order.acceptablePrice = eventData.getUintItem("acceptablePrice")!;
+  order.callbackGasLimit = eventData.getUintItem("callbakGasLimit");
+  order.minOutputAmount = eventData.getUintItem("minOutputAmount");
+  order.executionFee = eventData.getUintItem("executionFee")!;
+  order.updatedAtBlock = eventData.getUintItem("updatedAtBlock");
+  order.orderType = eventData.getUintItem("orderType")!;
+  order.isLong = eventData.getBoolItem("isLong");
+  order.shouldUnwrapNativeToken = eventData.getBoolItem(
+    "shouldUnwrapNativeToken"
+  )!;
+
+  let isFrozen = eventData.getBoolItem("isFrozen");
+
+  if (isFrozen) {
+    order.status = "Frozen";
+  } else {
+    order.status = "Created";
+  }
+
+  order.createdTxn = transaction.id;
+  order.save();
+
+  return order;
+}
+
+export function handleOrderCancelled(
+  eventData: EventData,
+  transaction: Transaction
+): Order {
+  let key = eventData.getBytes32Item("key")!.toHexString();
+
+  let order = Order.load(key);
+
+  if (order == null) {
+    throw new Error("Order not found " + key);
+  }
+
+  order.status = "Cancelled";
+  order.cancelledReason = eventData.getStringItem("reason");
```

Comment:
mmm let's store `reasonBytes` as well, i'm not sure `reason` will always contain readable value. on keeper side i decode `reasonBytes` because i wasn't able to retrieve readable reason in some cases 🤔